### ISSUE TITLE
Segfault about an hour into rmlint -D

### ISSUE THREAD

**Issue by jim-collier** [sahib/rmlint#346]
**Command**:

`rmlint "/mnt/btrfs/array-f7" --no-crossdev --no-followlinks  --see-symlinks --hidden -o progressbar -o summary --no-with-color --xattr-write --write-unfinished --algorithm=blake2bp -vvv --merge-directories --xattr-read --rank-by=pHMLDa -o csv -o json -o sh -o py`

**Console summary output**:

The progress bar and this text got all mixed together, but this seems to be the relevant stuff:

`Matching (223228 dupes of 111756 originals; 6109.70 GB to scan in 514238 files, ETA: 11d  8h  3m 38s)`
`ERROR: Aborting due to a fatal error. (signal received: Segmentation fault)`
`ERROR: Please file a bug report (See rmlint -h)`

**rmlint version**:

`rmlint --version`
`version 2.8.0 compiled: Jan 24 2019 at [23:31:58] "Maidenly Moose" (rev b4f1a33c)`
`compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support`

**Other version info**:

`Distribution version ...........: Ubuntu 18.04.1 LTS`
`Architectures supported ........: amd64, i386`
`Kernel version .................: 4.15.0-43-generic`
`CPU ............................: Intel(R) Core(TM) i7-4610M CPU @ 3.00GHz`
`Core count .....................: 4`
`Platform .......................: x86_64`
`Memory (KB).....................: 16329384`

I'm currently logging another run with -vvv for upload, which will take an hour or two before it hits this again (and includes this error). I can also run and log with gdb and valgrind if necessary.

**Comment by sahib** [comment]
Please run this with debug symbols (``scons DEBUG=1 GDB=1``) and ``gdb -ex r --args ./rmlint [your args]``

**Comment by jim-collier** [comment]
OK I recompiled with those options (`scons DEBUG=1 GDB=1`), and am currently running it via `gdb -ex r --args` . I'm not familiar with gdb but google says the last step will look something like `gdb /usr/bin/rmlint <coredump> -ex "thread apply all bt" -ex "quit" > gdb_rmlint_thread-apply-all-bt.log`. I'm assuming <coredump> will be a self-evident output file somewhere. :-)

**Comment by sahib** [comment]
That would work too probably, but the advantage of my command above is that you:

* Don't need a coredump.
* Have a gdb shell open where you can type "help" if something goes wrong.

**Comment by jim-collier** [comment]
The problem was I didn't really understand how gdb works. (I didn't know it dropped you into a dedicated shell if the program it launches terminates.) I understand better now.

With a debug build on the latest dev version, run via gdb, either it wasn't segfaulting any more, or was taking significantly longer to get to that point. After 10-ish hours, I had to kill it. I'll try running a regular build of the latest dev version, without gdb, and see if it does better. If not, I'm not sure where I can go from there, at least without building a test volume with as much data. (Because I'm worried about killing my drives by hammering on them for days on end to trap a segfault.)

And also, to try to fix the "Checksum too old for " problem in #347, I'm going to first clear the extended attributes, then recalc them with the default blake2b (instead of sha256 and then blake2bp which I tried). I'll see if that helps this issue.
